#!/usr/bin/env python
PACKAGE = "social_card_reader"
from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

gen.add("userDiameter",     double_t,    0, "User card white circle outer diameter [cm]", 5.0,  0.0, 50.0)
gen.add("masterDiameter",   double_t,    0, "Master card white circle outer diameter [cm]", 7.0,  0.0, 50.0)
gen.add("distanceTolerance",   double_t,    0, "Maximal discrepancy between RGB and depth distance estimate [%].", 20.0, 0.0, 100.0)

gen.add("initialCircularityTolerance",   double_t,    0, "Initial circularity test tolerance [%].", 30.0, 0.0, 100.0)
gen.add("finalCircularityTolerance",     double_t,    0, "Final circularity test tolerance [%].", 2.0,  0.0, 100.0)

gen.add("areaRatioTolerance",   double_t,    0, "Tolerance of black and white area ratios  [%].", 40.0,  0.0, 200.0)
gen.add("centerDistanceToleranceRatio",   double_t,    0, "Concentricity test ratio  [%].", 10.0,  0.0, 100.0)
gen.add("centerDistanceToleranceAbs",     double_t,    0, "Concentricity test absolute [px].", 5.0,  0.0, 25.0)

exit(gen.generate(PACKAGE, "social_card_reader", "social_card"))
